spacewalk/certs-tools/mgr_ssl_cert_setup.py

```diff
@@ -427,40 +425,44 @@ def generateCertWithChainFile(serverCert, workdir, certData):
     return retContent
 
 
-def deployApache(workdir):
+def deployApache(apache_cert_content, server_key_content):
     if os.path.exists(APACHE_KEY_FILE):
         os.remove(APACHE_KEY_FILE)
     if os.path.exists(APACHE_CRT_FILE):
         os.remove(APACHE_CRT_FILE)
-    shutil.copy(os.path.join(workdir, SRV_KEY_NAME), APACHE_KEY_FILE)
-    if os.path.exists(os.path.join(workdir, APACHE_CRT_NAME)):
-        shutil.copy(os.path.join(workdir, APACHE_CRT_NAME), APACHE_CRT_FILE)
+    with open(APACHE_KEY_FILE, "w") as f:
+        f.write(server_key_content)
+    if apache_cert_content:
```

Comment:
I think this "if" test is not needed. We checked already before that this cannot be empty.